codesign/lib/src/file_codesign_visitor.dart

```diff
@@ -164,4 +183,51 @@ class FileCodesignVisitor {
       processManager: processManager,
     );
   }
+
+  /// Visit and codesign a binary with / without entilement.
+  ///
+  /// At this stage, the virtual [entitlementCurrentPath] accumulated through the recursive visit, is compared
+  /// against the paths extracted from [fileWithEntitlements], to help determine if this file should be signed
+  /// with entitlements.
+  Future<void> visitBinaryFile({required FileSystemEntity binaryEntity, required String entitlementParentPath}) async {
+    if (await isSymlink(binaryEntity.absolute.path, processManager)) {
+      return;
+    }
+
+    final String currentFileName = binaryEntity.absolute.path.split('/').last;
+    final String entitlementCurrentPath = '$entitlementParentPath/$currentFileName';
+
+    if (!fileWithEntitlements.contains(entitlementCurrentPath) &&
+        !fileWithoutEntitlements.contains(entitlementCurrentPath)) {
+      log.severe('The system has detected a binary file at $entitlementCurrentPath\n'
+          'but it is not in the entitlements configuartion files your provided \n'
+          'if this is a new engine artifact, please add it to one of the entilements.txt files\n');
+      throw CodesignException(fixItInstructions);
+    }
+    log.info('\n signing file at path ${binaryEntity.absolute.path}');
```

Comment:
why the leading `'\n '` in all these info logs?

Reply:
True, log already has formatting. Removed all log related \n.